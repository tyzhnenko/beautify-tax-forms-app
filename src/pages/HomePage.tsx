import { Typography } from "@material-tailwind/react";
import { Link } from "react-router-dom";
import useTitle from "../utils/useTitle";

function HomePage() {
  useTitle("Головна сторінка");

  return (
    <div className="container flex flex-col justify-between">
      <div className="pt-15">
        <Typography as="article">
          <h1 className="text-4xl text-center pb-10">Нормальні форми ДФС</h1>
          <p className="pt-5">
            Цей проект присвячиний для того щоб дати можливість працювати з
            формами які ДФС України надає у XML форматах.
          </p>
          <p className="pt-5">
            Якщо у вас колись виникали складнощі в перенесені данних з форм які
            надає ДФС до табличних редакторів. Ласкаво просимо.
          </p>
          <p className="pt-5">
            Список форм/звітів які підтримуються можна подивитись на сторінці{" "}
            <Link className="underline" to="/forms">
              форм
            </Link>
          </p>
          <p className="pt-5">
            Усі данні обробляються <i>у вас в браузері</i>. Цей проект не
            зберігає жодних данних на стороні сервера.
          </p>
          <p className="pt-5">
            На поекті використувється система Google Analitics та їх cookie
            ідентифікатори.
          </p>
        </Typography>
      </div>
    </div>
  );
}

export default HomePage;
