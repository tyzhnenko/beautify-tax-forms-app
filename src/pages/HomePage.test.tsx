/**
 * @jest-environment jsdom
 */

import { render, screen } from "@testing-library/react";
import HomePage from "./HomePage";
import "@testing-library/jest-dom";
import { BrowserRouter } from "react-router-dom";

describe("HomePage component", () => {
  it("should render the component", () => {
    render(
      <BrowserRouter>
        <HomePage />
      </BrowserRouter>
    );
    expect(document.title).toEqual("Головна сторінка");
    expect(screen.getByText(/Нормальні форми ДФС/i)).toBeInTheDocument();
  });

  it("should match snapshot", () => {
    const { container } = render(
      <BrowserRouter>
        <HomePage />
      </BrowserRouter>
    );
    expect(container).toMatchSnapshot();
  });
});
