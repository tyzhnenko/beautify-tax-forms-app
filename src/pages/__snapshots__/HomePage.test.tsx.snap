// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomePage component should match snapshot 1`] = `
<div>
  <div
    class="container flex flex-col justify-between"
  >
    <div
      class="pt-15"
    >
      <article
        class="block antialiased font-sans text-base font-light leading-relaxed text-inherit"
      >
        <h1
          class="text-4xl text-center pb-10"
        >
          Нормальні форми ДФС
        </h1>
        <p
          class="pt-5"
        >
          Цей проект присвячиний для того щоб дати можливість працювати з формами які ДФС України надає у XML форматах.
        </p>
        <p
          class="pt-5"
        >
          Якщо у вас колись виникали складнощі в перенесені данних з форм які надає ДФС до табличних редакторів. Ласкаво просимо.
        </p>
        <p
          class="pt-5"
        >
          Список форм/звітів які підтримуються можна подивитись на сторінці
           
          <a
            class="underline"
            href="/forms"
          >
            форм
          </a>
        </p>
        <p
          class="pt-5"
        >
          Усі данні обробляються 
          <i>
            у вас в браузері
          </i>
          . Цей проект не зберігає жодних данних на стороні сервера.
        </p>
        <p
          class="pt-5"
        >
          На поекті використувється система Google Analitics та їх cookie ідентифікатори.
        </p>
      </article>
    </div>
  </div>
</div>
`;
